src/main/java/org/spongepowered/common/registry/type/item/ItemStackComparatorRegistryModule.java

```diff
@@ -0,0 +1,115 @@
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.common.registry.type.item;
+
+import com.google.common.collect.Lists;
+import org.spongepowered.api.data.Property;
+import org.spongepowered.api.data.manipulator.DataManipulator;
+import org.spongepowered.api.data.manipulator.mutable.entity.DamageableData;
+import org.spongepowered.api.item.inventory.ItemStack;
+import org.spongepowered.api.item.inventory.ItemStackComparators;
+import org.spongepowered.api.registry.RegistryModule;
+import org.spongepowered.api.registry.util.RegisterCatalog;
+
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import javax.annotation.Nullable;
+
+@SuppressWarnings("unchecked")
+public final class ItemStackComparatorRegistryModule implements RegistryModule {
+
+    @RegisterCatalog(ItemStackComparators.class)
+    private final Map<String, Comparator<ItemStack>> comparators = new HashMap<>();
+
+    @Override
+    public void registerDefaults() {
+
+        Comparator<ItemStack> type = Comparator.comparing(i -> i.getType().getId());
+        this.comparators.put("TYPE", type);
+        Comparator<ItemStack> size = Comparator.comparing(ItemStack::getQuantity);
+        this.comparators.put("SIZE", size);
+        Comparator<ItemStack> typeSize = type.thenComparing(size);
+        this.comparators.put("TYPE_SIZE", typeSize);
+        this.comparators.put("DEFAULT", typeSize);
+        Properties properties = new Properties();
+        this.comparators.put("PROPERTIES", properties);
+        ItemDataComparator itemData = new ItemDataComparator();
+        this.comparators.put("ITEM_DATA", itemData);
+        this.comparators.put("ITEM_DATA_IGNORE_DAMAGE", new ItemDataComparator(DamageableData.class));
+        this.comparators.put("IGNORE_SIZE", type.thenComparing(properties).thenComparing(itemData));
+        this.comparators.put("ALL", type.thenComparing(size).thenComparing(properties).thenComparing(itemData));
+    }
+
+    private static final class Properties implements Comparator<ItemStack> {
+
+        @Override
+        public int compare(ItemStack o1, ItemStack o2) {
+            List<Property<?, ?>> properties = Lists.newArrayList(o2.getApplicableProperties());
+            for (Property<?, ?> property : o1.getApplicableProperties()) {
+                if (properties.contains(property)) {
```

Comment:
Since you call `contains` and `remove`, I'd say you should change `properties` to a `Set`.